### ISSUE TITLE
[Accessibility] "Recently" is displayed on two rows when "Large text" is set to maximum

### ISSUE THREAD

**Issue by SimonBasca** [mozilla-mobile/firefox-ios#10923]
## Steps to reproduce
1. Increase Large text to maximum
1. Have a few items in "Recently saved" and "Recently visited"
1. Observe the titles
### Expected behavior

* "Recently" should be displayed on one row
### Actual behavior
* "Recently" is displayed on two rows
### Device & build information
* Device: iPhone 12 Pro 15.4
* Build version: 176a0f8578a78e652407b322e5bf2176e616e636
### Notes
Attachments: 

![Screenshot](https://user-images.githubusercontent.com/32130829/172159219-22874b30-3e5b-4198-a59a-96487d569506.jpeg)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-4335)

**Comment by lmarceau** [comment]
An idea to fix this would be to have header layout change so that the "Show all" button is under the title when we reach bigger font sizes.

**Comment by SimonBasca** [comment]
@lmarceau is there a way to limit the font to a certain threshold where "Recently"  will be displayed on one row?

https://user-images.githubusercontent.com/32130829/172164039-784bfd39-9b58-48fb-9a28-31a500c9280a.mp4

**Comment by lmarceau** [comment]
In theory, yes, but in practice limiting font size is not really practical. Reasons are that we shouldn't limit font size since it's a user preference. Some users might depend on being able to read at those font size. Also if we consider that we have different languages and different form factors for phones, it becomes impossible to know in all cases that the title would fit and not be layout on more than 1 line. Adjusting the layout to have more space for bigger font sizes to layout properly makes more sense IMO.

**Comment by anna-yeddi** [comment]
The described behavior is actually accessible: it does allow for the text to reflow without causing any two-dimensional scrolling or without text being cut or lost - thus, satisfying both related WCAG 2.1 Success Criteria: [1.4.10 Reflow](https://www.w3.org/WAI/WCAG21/Understanding/reflow.html) and [1.4.4 Resize text](https://www.w3.org/WAI/WCAG21/Understanding/resize-text.html). Limiting user's ability to resize text would create a major accessibility issue.

But I agree, reflowing the `Show all` link to the next row allowing the title to take over more space would improve the existent behavior for users with low vision, thus access-s4

**Comment by data-sync-user** [comment]
➤ Winnie Teichmann commented:

Alina Moldovan did you have a chance to verify if this bug is still reproducible? Otherwise we should close it.

**Comment by data-sync-user** [comment]
➤ Alina Moldovan commented:

Validated this issue using iPhone 6s (15.7), iPhone 15+ (18.2), iPhone 16 Pro (18.2)

"Recently saved" section was renamed as Bookmark and Recently visited section was removed. 

Checked the other elements on the homepage and didn’t observed any inconsistency: 

[^ScreenRecording_02-10-2025 19-52-50_1.MP4]

[^ScreenRecording_02-10-2025 19-49-31_1.MP4]

[^ScreenRecording_02-10-2025 19-14-29_1.MP4]

**Comment by data-sync-user** [comment]
➤ Winnie Teichmann commented:

Thanks for testing this, Alina Moldovan . Could you please close the issue or mark it as QA verified?

**Comment by data-sync-user** [comment]
➤ Alina Moldovan commented:

Sure, thank you!
